.github/workflows/go-test.yml

```diff
@@ -0,0 +1,31 @@
+---
+# Test built docker images by building simple projects inside them
+
+name: go test
+on:
+  pull_request:
+  merge_group:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  go-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
Copy-paste error, should have been `fetch-depth: 1` (which is the default)

Reply:
If I am not mistaken, we only need `fetch-depth: 0` in `lint.yml/commit-lint`, and everywhere else this can be removed. Maybe also `lint.yml/megalinter` but I am not sure about that

Reply:
Currently only commit lint requires it as it needs access to the whole git history. Other candidates would be automatic release tools which need to see the whole git log in order to generate changelogs or tags. Megalinter shouldn't need it as well as it just operates on the code and not the git logs.